databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,501 @@
+import base64
+import collections.abc
+import json
+import os
+import uuid
+from pathlib import Path
+from typing import Dict, List
+
+import dask.array
+import h5py
+import numpy
+import pandas
+import pydantic
+import pymongo
+
+from fastapi import APIRouter, HTTPException, Request, Security
+from tiled.adapters.array import ArrayAdapter
+from tiled.adapters.dataframe import DataFrameAdapter
+from tiled.adapters.utils import IndexersMixin, tree_repr
+from tiled.server.core import json_or_msgpack
+from tiled.server.dependencies import entry
+from tiled.query_registration import QueryTranslationRegistry
+
+from tiled.structures.core import StructureFamily
+from tiled.structures.dataframe import deserialize_arrow, serialize_arrow
+
+from tiled.server.pydantic_array import ArrayStructure
+from tiled.server.pydantic_dataframe import DataFrameStructure
+from tiled.utils import APACHE_ARROW_FILE_MIME_TYPE, UNCHANGED
+
+from typing import Union
+
+from .schemas import Document
+from .queries import RawMongo
+
+from sys import platform
+
+
+class PostMetadataRequest(pydantic.BaseModel):
+    structure_family: StructureFamily
+    structure: Union[ArrayStructure, DataFrameStructure]
+    metadata: Dict
+    specs: List[str]
+
+
+class PostMetadataResponse(pydantic.BaseModel):
+    uid: str
+
+
+router = APIRouter()
+
+
+@router.post("/node/metadata/{path:path}", response_model=PostMetadataResponse)
+def post_metadata(
+    request: Request,
+    body: PostMetadataRequest,
+    entry=Security(entry, scopes=["write:data", "write:metadata"]),
+):
+    if not isinstance(entry, MongoAdapter):
```

Comment:
this is too restrictive for the generic case. I think the pythonic thing to do would be to just call `entry.post_metadata` and catch the error if the method does not exist.